play/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PublishSubscriptions.kt

```diff
@@ -49,31 +49,52 @@ internal abstract class PublishSubscriptions @Inject constructor(
 
     @TaskAction
     fun publishSubscriptions(changes: InputChanges) {
+
         changes.getFileChanges(subscriptionsDir)
                 .filterNot { it.changeType == ChangeType.REMOVED }
                 .filter { it.fileType == FileType.FILE }
+                .map {
+                    if (it.file.name.endsWith(".$SUBSCRIPTION_METADATA_SUFFIX.json")) {
+                        it.file.parentFile.resolve(
+                                it.file.name.replace(
+                                        "\\.$SUBSCRIPTION_METADATA_SUFFIX\\.json$".toRegex(), ".json"))
```

Comment:
Pull `".$SUBSCRIPTION_METADATA_SUFFIX.json"` out into a local variable and make this `it.file.name.dropLast(metadataFileSuffix.length) + ".json"`.

Also add a comment explaining that we need these potential duplicates so that if someone makes an incremental change to just the metadata file, we'll still notice that and try to publish the updated subscription.